src/analyses/goto_check.cpp

```diff
@@ -907,23 +904,57 @@ void goto_checkt::pointer_overflow_check(
   if(!enable_pointer_overflow_check)
     return;
 
-  if(expr.id()==ID_plus ||
-     expr.id()==ID_minus)
+  if(expr.id()!=ID_plus &&
+     expr.id()!=ID_minus)
+    return;
+
+  if(expr.operands().size()>2)
   {
-    if(expr.operands().size()==2)
+    // The overflow checks are binary!
+    // We break these up.
+
+    for(unsigned i=1; i<expr.operands().size(); i++)
     {
-      exprt overflow("overflow-"+expr.id_string(), bool_typet());
-      overflow.operands()=expr.operands();
+      exprt tmp;
 
-      add_guarded_claim(
-        not_exprt(overflow),
-        "pointer arithmetic overflow on "+expr.id_string(),
-        "overflow",
-        expr.find_source_location(),
-        expr,
-        guard);
+      if(i==1)
+        tmp=expr.op0();
+      else
+      {
+        tmp=expr;
+        tmp.operands().resize(i);
```

Comment:
You may want to make it explicit here that you are chopping off the last nodes for now to do an intermediate check